File: backend/app/schemas/transcript.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class TranscriptChunkBase(BaseModel):
    speaker: str = Field(min_length=1, max_length=100)
    content: str = Field(min_length=1)


class TranscriptChunkUpdate(BaseModel):
    speaker: str | None = Field(default=None, min_length=1, max_length=100)
    content: str | None = Field(default=None, min_length=1)


class TranscriptChunkRead(TranscriptChunkBase):
    id: uuid.UUID
    meeting_id: uuid.UUID
    start: float = Field(ge=0)
    end: float = Field(ge=0)
    confidence: float | None = Field(default=None, ge=0, le=1)
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class TranscriptListRead(BaseModel):
    meeting_id: uuid.UUID
    chunks: list[TranscriptChunkRead]

File: frontend/meetingPilot-frontend/src/types/transcript.ts
export interface TranscriptChunkBase {
  speaker: string
  content: string
}

export interface TranscriptChunkUpdate {
  speaker?: string | null
  content?: string | null
}

export interface TranscriptChunkRead extends TranscriptChunkBase {
  id: string
  meeting_id: string
  start: number
  end: number
  confidence: number | null
  updated_at: string | null
}

export interface TranscriptListRead {
  meeting_id: string
  chunks: TranscriptChunkRead[]
}
